metadata-ingestion/tests/unit/test_teradata_source.py

```diff
@@ -1494,3 +1496,267 @@ def _create_mock_entry(
         entry.default_database = default_database
         entry.session_id = session_id or f"session_{query_id}"
         return entry
+
+
+class TestTeradataRegressionFixes:
+    """Test suite for regression fixes in Teradata source.
+
+    This class tests the fixes for two critical issues:
+    1. Time window defaults not being applied correctly
+    2. Missing incremental lineage support
+    """
```

Comment:
I wouldn't wrap these tests in a `TestTeradataRegressionFixes` and just add the new tests in the corresponding  test classe among the existing ones.